### PR TITLE
Make init method have valid jsdoc

### PR DESCRIPTION
**Pull request by JakeChampion**
Documented the return type and made the argument names match the documented parameter names

### CODE REVIEW

File: templates/src-js.js [hunk 1]
```diff
@@ -43,21 +43,22 @@ module.exports = answers => {
 		}, {});
 	}
 	/**
-	 * Initialise ${name} component.
+	 * Initialise o-example component/s.
```
Comment (notlee) [inline templates/src-js.js:46]: was this intentional? 
```suggestion
	 * Initialise ${name} component/s.
```

File: templates/src-js.js [hunk 2]
```diff
@@ -43,21 +43,22 @@ module.exports = answers => {
...
 	 * @param {(HTMLElement|String)} rootElement - The root element to intialise the component in, or a CSS selector for the root element
 	 * @param {Object} [options={}] - An options object for configuring the component
+	 * @returns {Example|Example[]} The newly constructed Example components
```
Comment (notlee) [inline templates/src-js.js:49]: This `Example` should use the given component name
```suggestion
	 * @returns {Example|Example[]} The newly constructed Example components
```

File: templates/src-js.js (hunk 1 above)
Comment (JakeChampion) [reply templates/src-js.js:46]: yeah, I thought it needed the `/s` as it does either initialise one or many components

Comment (notlee) [reply templates/src-js.js:46]: sorry I meant `o-example` instead of `${name}`

File: templates/src-js.js (hunk 2 above)
Comment (JakeChampion) [reply templates/src-js.js:49]: Yes 🤦 I copy pasted from the tutorial which makes an Example component

File: templates/src-js.js [hunk 3]
```diff
@@ -43,21 +43,22 @@ module.exports = answers => {
...
+			rootElement = document.body;
 		}
-		if (!(rootEl instanceof HTMLElement)) {
-			rootEl = document.querySelector(rootEl);
+		if (!(rootElement instanceof HTMLElement)) {
+			rootElement = document.querySelector(rootElement);
 		}
-		if (rootEl instanceof HTMLElement && rootEl.matches('[data-o-component=${name}]')) {
-			return new ${className}(rootEl, opts);
+		if (rootElement instanceof HTMLElement && rootElement.matches('[data-o-component=o-example]')) {
```
Comment (JakeChampion) [inline templates/src-js.js:58]:
```suggestion
		if (rootElement instanceof HTMLElement && rootElement.matches('[data-o-component=${name}]')) {
```

File: templates/src-js.js [hunk 4]
```diff
@@ -43,21 +43,22 @@ module.exports = answers => {
...
+			return new Example(rootElement, options);
```
Comment (JakeChampion) [inline templates/src-js.js:59]:
```suggestion
			return new ${className}(rootElement, options);
```

File: templates/src-js.js [hunk 5]
```diff
@@ -43,21 +43,22 @@ module.exports = answers => {
...
 		}
-		return Array.from(rootEl.querySelectorAll('[data-o-component="${name}"]'), rootEl => new ${className}(rootEl, opts));
+		return Array.from(rootElement.querySelectorAll('[data-o-component="o-example"]'), rootEl => new Example(rootEl, options));
```
Comment (JakeChampion) [inline templates/src-js.js:61]:
```suggestion
		return Array.from(rootElement.querySelectorAll('[data-o-component="${name}"]'), rootEl => new ${className}(rootEl, options));
```

**Comment by JakeChampion** [comment]
:tada: This PR is included in version v0.2.12 :tada:

**Pull request merged**